src/wrappers/themis/Obj-C/objcthemis/scell_seal.m

```diff
@@ -60,18 +64,15 @@ - (NSData *)wrapData:(NSData *)message context:(NSData *)context error:(NSError
         return nil;
     }
 
-    NSData * wrappedData = [[NSData alloc] initWithBytes:wrappedMessage length:wrappedMessageLength];
-    free(wrappedMessage);
-    return wrappedData;
-
+    return [NSData dataWithBytesNoCopy:wrappedMessage length:wrappedMessageLength];
```

Comment:
https://developer.apple.com/reference/foundation/nsdata/1547229-datawithbytesnocopy

The returned object takes ownership of the bytes pointer and frees it on deallocation. Therefore, bytes must point to a memory block allocated with malloc.